docs/source/tutorials/installation.rst

```diff
@@ -0,0 +1,148 @@
+
+.. note::
+    If you use zsh, ``pip install .[full]`` may not work properly. In this case,
+    try installing the library via ``pip install ".[full]"``
+
+PyPI Install
+~~~~~~~~~~~~
+
+.. warning::
+
+    PyPI install is currently not stable, and we advise the users to use the
+    local install instead. With future otx versions, this feature will be more
+    stable, such that the user could install the library via,
+
+.. code-block:: bash
+
+    pip install otx[full]
+
+
+Options
+~~~~~~~
+Unlike the well-known libraries and frameworks, ``otx`` provides number of
+task types such as classification, detection, segmentation and anomaly in a
+single library. Even though this is a rather unique feature, some users may want
+to install a specific task instead. ``otx`` therefore makes this installation
+completely optional, depending on the algorithmic needs.
+
+Basic Installation
+""""""""""""""""""
+This option is the most basic option, where pip only installs ``otx`` api and
+cli modules. For those who are only interested in the api and cli capabilities
+could install this option.
+
+.. code-block:: bash
+
+    # Install API and CLI only.
+    pip install otx
+
+Classification Task Installation
+""""""""""""""""""""""""""""""""
+Classification option would install ``otx`` with classification algorithms only.
+
+.. code-block:: bash
+
+    # Install only otx classification library
+    pip install otx[classification]
+
+Detection Task Installation
+"""""""""""""""""""""""""""
+Detection would install the ``otx`` with detection algorithms, which are based
+on `mmdetection <https://github.com/open-mmlab/mmdetection>`_ library but with a
+lot more features.
+
+.. code-block:: bash
+
+    # Install otx detection library
+    pip install otx[detection]
+
+Segmentation Task Installation
+""""""""""""""""""""""""""""""
+Similar to the detection task, ``segmentation`` option would install segmentation
+algorithms that utilizes `mmsegmentation <https://github.com/open-mmlab/mmsegmentation>`_
+library, but again with more end-to-end functionality such as OpenVINO export.
+
+.. code-block:: bash
+
+    # Install segmentation library
+    pip install otx[segmentation]
+
+Anomaly Task Installation
+"""""""""""""""""""""""""
+``anomaly`` option would install the anomaly classification, detection and
+segmentation tasks that uses `anomalib <https://github.com/openvinotoolkit/anomalib>`_.
+Users would train models within a one-class classification fashion by utilizing
+only the ``normal`` class during training to find any ``abnormality`` during the
+validation, test or inference.
+
+.. code-block:: bash
+
+    # Install otx anomaly library
+    pip install otx[anomaly]
+
+Full Installation
+"""""""""""""""""
+``full``` option would install all of the tasks together, overall making ``otx``
+a complete library that a user could train a supervised, semi-supervised or
+self-supervised classification, detection or segmentation algorithms with full
```

Comment:
self-supervised is not intended to be added in Q4 and early Q1 at least. Should we mention this? Or is it just something global for the future?

Reply:
I think we should mention it. e.g. self-supervised classification(planned to be added in Q2'23)